### DISCUSSION TITLE
Zigbee2MQTT Start at Boot

### DISCUSSION THREAD

**Discussion by xtremeideaz** [tteck/Proxmox#99] [Q&A] [answered]
Hello,

Thank you for this wonderful lxc script. 
How do i get the Zigbee2MQTT to  start up automatically without having to insert the CD start command every time there's a system reboot?

Please help.

**Comment by tteck** [comment, marked as answer]
If you installed between 2/23/2022 and 3/19/2022 run  `systemctl enable zigbee2mqtt.service` in the LXC console. I forgot to enable the service when I wrote the V2 script
Before 2/23/2022 and after 3/19/2022 it should start at boot

**Reply by xtremeideaz** [reply]
Thanks. i tried it but it's not working. when enter that line i get the following error.

> zigbee2mqtt@1.24.0 start /opt/zigbee2mqtt
> node index.js

Zigbee2MQTT:info  2022-03-27 08:56:58: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-03-27.08-56-58' filename: log.txt
Zigbee2MQTT:info  2022-03-27 08:56:58: Starting Zigbee2MQTT version 1.24.0 (commit #7a2ddf24)
Zigbee2MQTT:info  2022-03-27 08:56:58: Starting zigbee-herdsman (0.14.20)
CREATED DECONZ ADAPTER
Zigbee2MQTT:error 2022-03-27 08:56:58: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-03-27 08:56:58: Failed to start zigbee
Zigbee2MQTT:error 2022-03-27 08:56:58: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-03-27 08:56:58: Exiting...
Zigbee2MQTT:error 2022-03-27 08:56:58: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/deconz/driver/driver.ts:182:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.24.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.24.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-03-27T08_56_58_699Z-debug.log
root@zigbee2mqtt:/opt/zigbee2mqtt#

**Reply by xtremeideaz** [reply]
However,  when i disable it, reboot and end the cd start command, everything works just fine. 

Should i delete and just reinstall?

**Reply by tteck** [reply]
try `systemctl stop zigbee2mqtt` then `systemctl enable zigbee2mqtt.service` and last `systemctl start zigbee2mqtt` before reinstalling

**Reply by tteck** [reply]
there'a also a `z2m-copy-data.sh` script to copy your data to a new LXC
Just don't run both at the same time

**Reply by tteck** [reply]
Usually when I see `Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'` Z2M is already running

**Reply by tteck** [reply]
always try loading the frontend

**Reply by xtremeideaz** [reply]
> try `systemctl stop zigbee2mqtt` then `systemctl enable zigbee2mqtt.service` and last `systemctl start zigbee2mqtt` before reinstalling

This did the trick. Thanks. It's working now.

**Reply by xtremeideaz** [reply]
> there'a also a `z2m-copy-data.sh` script to copy your data to a new LXC Just don't run both at the same time

Thanks. i'll write this down for future use. Thanks a lot.
